File: solutions/typescript/secret-handshake/1/secret-handshake.ts
export function commands(command: number): string[] {
    // convert command to binary
    //
    const binary = command.toString(2);
    const commands = [];
    if (binary.length >= 1 && binary[binary.length - 1] === '1') {
        commands.push('wink');
    }
    if (binary.length >= 2 && binary[binary.length - 2] === '1') {
        commands.push('double blink');
    }
    if (binary.length >= 3 && binary[binary.length - 3] === '1') {
        commands.push('close your eyes');
    }
    if (binary.length >= 4 && binary[binary.length - 4] === '1') {
        commands.push('jump');
    }
    if (binary.length >= 5 && binary[binary.length - 5] === '1') {
        commands.reverse();
    }
    return commands;
}

File: solutions/go/secret-handshake/1/secret_handshake.go
package secret

import (
	"fmt"
	"strconv"
)

func Handshake(code uint) []string {
	binary := strconv.FormatInt(int64(code), 2)
	paddedBinary := fmt.Sprintf("%05s", binary)
	reversed := reverse(paddedBinary)
	handshake := []string{}
	if reversed[0] == '1' {
		handshake = append(handshake, "wink")
	}
	if reversed[1] == '1' {
		handshake = append(handshake, "double blink")
	}
	if reversed[2] == '1' {
		handshake = append(handshake, "close your eyes")
	}
	if reversed[3] == '1' {
		handshake = append(handshake, "jump")
	}
	if reversed[4] == '1' {

		handshake = reverseSlice(handshake)
	}

	return handshake
}

func reverseSlice(handshake []string) []string {
	for i, j := 0, len(handshake)-1; i < j; i, j = i+1, j-1 {
		handshake[i], handshake[j] = handshake[j], handshake[i]
	}
	return handshake
}

func reverse(binary string) string {
	runes := []rune(binary)
	for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
		runes[i], runes[j] = runes[j], runes[i]
	}
	return string(runes)
}
